test/python/algorithms/optimizers/test_spsa.py

```diff
@@ -188,7 +188,7 @@ def test_estimate_stddev(self, max_evals_grouped):
         See https://github.com/Qiskit/qiskit-nature/issues/797"""
 
         def objective(x):
-            if len(x.shape == 2):
+            if x.shape == 2:
```

Comment:
if `len(x.shape) == 2`? shape is a tuple right. It might be a single value in the tuple if its a single dimensional array if this is what its trying to differentiate

Reply:
whoops my bad, thanks!